File: kata-setup-typescript/src/Item.AgedBrie.ts
import { Item, Itemer } from "./Item.js"

export class AgedBrie extends Item implements Itemer {
  constructor(sellIn: number, quality: number) {
    super("Aged Brie", sellIn, quality)
  }

  updateQuality(newSellIn: number): void {
    if (this.sellIn < 0) {
      this.quality += 2
    } else {
      this.quality++
    }
  }
}

File: kata-setup-java/src/main/java/org/example/ItemAgedBrie.java
package org.example;

public class ItemAgedBrie extends Item {

  public ItemAgedBrie(int sellIn, int quality) {
    super("Aged Brie", sellIn, quality);
  }

  @Override
  public void updateQuality() {
    if (hasPassedSellIn()) {
      quality = quality + 2;
    } else {
      quality = quality + 1;
    }
    if (quality > 50) {
      quality = 50;
    }
  }
}
